crates/provider-keyvalue-redis/src/lib.rs

```diff
@@ -394,6 +713,28 @@ impl Provider for KvRedisProvider {
         Ok(())
     }
 
+    #[instrument(level = "info", skip_all, fields(target_id = info.get_target_id()))]
+    async fn delete_link_as_source(&self, info: impl LinkDeleteInfo) -> anyhow::Result<()> {
+        let component_id = info.get_target_id();
+        let mut aw = self.sources.write().await;
+        aw.retain(|(src_id, _link_name), _| src_id != component_id);
```

Comment:
Perhaps this was before we added this, but we do include the `link_name` now in the LinkDeleteInfo! So you should be able to just `remove` the key here